### PR TITLE
test: Migrate tests to temp folders

### PR DESCRIPTION
**Pull request by MichaelsJP**
### Pull Request Checklist

- [ ] 1. I have [**rebased**][rebase] the latest version of the master branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes # .

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/installation/Configuration.html
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/master/CONTRIBUTE.md#pull-request-guidelines

### CODE REVIEW

File: ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java [hunk 1]
```diff
@@ -16,16 +16,16 @@ void setUp(){
     }
     @ParameterizedTest
     @CsvSource(value = {
-            "https://example.com/test-repo/planet,      https://example.com, test-repo, planet",
-            "https://example.com/test-repo/planet/,     https://example.com, test-repo, planet",
-            "' https://example.com/test-repo/planet ',  https://example.com, test-repo, planet",
-            "' https://example.com/test-repo/planet/ ', https://example.com, test-repo, planet",
-            "' http://example.com/test-repo/planet/ ',  http://example.com, test-repo, planet"
+            "https://example.com/foo1/foo2/foo3/test-repo,      https://example.com, test-repo",
```
Comment (jhaeu) [inline ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java]: suggested change:
``` ```

File: ors-api/src/main/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListener.java [hunk 2]
```diff
@@ -102,22 +104,27 @@ record SourceFileElements(String repoBaseUrlString, String repoName, String loca
...
-            LOGGER.debug("Configuration property 'source_file' contains a URL, using value as URL for a graphs repository");
-            sourceFilePropertyValue = sourceFilePropertyValue.trim().replaceAll("/$", "");
-            String[] urlElements = sourceFilePropertyValue.split("/");
+            URI uri = new URI(sourceFilePropertyValue);
+            accessScheme = uri.getScheme();
+            repoBaseUrlString = uri.getHost();
+            if (accessScheme == null || repoBaseUrlString == null) {
+                throw new URISyntaxException(sourceFilePropertyValue, "URI does not contain a valid schema or host");
+            }
+            repoBaseUrlString = accessScheme + "://" + repoBaseUrlString;
```
Comment (jhaeu) [inline ors-api/src/main/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListener.java:116]: this won't work, you cut off too much of the configured url - see my comment for the test for this method

File: ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java (hunk 1 above)
Comment (jhaeu) [inline ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java]: Please add this test and change the code to make it pass:
```suggestion
            "https://test-repo.openrouteservice.org/service/rest/ors-graphs-fastisochrones/, https://test-repo.openrouteservice.org/service/rest, ors-graphs-fastisochrones",
```

File: ors-engine/src/test/java/org/heigit/ors/routing/graphhopper/extensions/manage/ORSGraphFileManagerTest.java [hunk 3]
```diff
@@ -37,9 +37,9 @@ class ORSGraphFileManagerTest {
     private static final String GRAPHS_COVERAGE = "planet";
     private static final String GRAPHS_VERSION = "1";
     private static final String VEHICLE = "car";
-    private static String LOCAL_PATH = "src/test/resources/graphs";
     @TempDir(cleanup = CleanupMode.ON_SUCCESS)
-    Path tempDir;
+    private static Path TEMP_DIR;
```
Comment (jhaeu) [inline ors-engine/src/test/java/org/heigit/ors/routing/graphhopper/extensions/manage/ORSGraphFileManagerTest.java:41]: very cool, much better :)

File: ors-engine/src/test/java/org/heigit/ors/routing/graphhopper/extensions/manage/ORSGraphFileManagerTest.java [hunk 4]
```diff
@@ -95,6 +95,26 @@ void setupORSGraphManager(String hash, EngineConfig engineConfig) {
         orsGraphRepoManager.setFileManager(orsGraphFileManager);
     }
 
+    void setupORSGraphManager(String hash, String customGraphFolder) {
```
Comment (jhaeu) [inline ors-engine/src/test/java/org/heigit/ors/routing/graphhopper/extensions/manage/ORSGraphFileManagerTest.java]: much redundancy to the previous method - why not e.g. add a method 
```suggestion
    void setupORSGraphManager(String hash, EngineConfig engineConfig, String customGraphFolder)
```

**Review by jhaeu — CHANGES_REQUESTED** [review]
using temp folder is cool,
please fix the url parsing method

File: ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java (hunk 1 above)
Comment (MichaelsJP) [reply ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java]: Sure. I'll need to add it myself. GH can't

File: ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java [hunk 5]
```diff
@@ -16,16 +17,24 @@ void setUp(){
     }
     @ParameterizedTest
     @CsvSource(value = {
-            "https://example.com/test-repo/planet,      https://example.com, test-repo, planet",
-            "https://example.com/test-repo/planet/,     https://example.com, test-repo, planet",
-            "' https://example.com/test-repo/planet ',  https://example.com, test-repo, planet",
-            "' https://example.com/test-repo/planet/ ', https://example.com, test-repo, planet",
-            "' http://example.com/test-repo/planet/ ',  http://example.com, test-repo, planet"
+            "https://example.com/foo1/foo2/foo3/test-repo,      https://example.com, test-repo, true",
```
Comment (jhaeu) [inline ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java]: first line here has the wrong expectation, expected url is https://example.com/foo1/foo2/foo3
```suggestion
            "https://example.com/foo1/foo2/foo3/test-repo,      "https://example.com/foo1/foo2/foo3, test-repo, true",
```

File: ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java [hunk 6]
```diff
@@ -16,16 +17,24 @@ void setUp(){
...
+            "https://example.com/test-repo/,     https://example.com, test-repo, true",
+            "https://example.com/test-repo,  https://example.com, test-repo, true",
+            "' https://example.com/test-repo ',  https://example.com, test-repo, true",
+            "https://example.com/test-repos/, https://example.com, test-repos, true",
+            "http://example.com/test-repo/,  http://example.com, test-repo, true",
+            "/foo/test-repo/,  null, null, false"
     })
-    void extractSourceFileElements_withURL(String url, String expectedBaseUrl, String expectedRepoName, String expectedCoverage){
+    void extractSourceFileElements_withURL(String url, String expectedBaseUrl, String expectedRepoName, Boolean isUrl){
         ORSInitContextListener.SourceFileElements elements = orsInitContextListener.extractSourceFileElements(url);
```
Comment (jhaeu) [inline ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java:29]: good, thanks for testing the file case also

File: ors-engine/src/test/java/org/heigit/ors/routing/graphhopper/extensions/manage/ORSGraphFileManagerTest.java (hunk 4 above)
Comment (MichaelsJP) [reply ors-engine/src/test/java/org/heigit/ors/routing/graphhopper/extensions/manage/ORSGraphFileManagerTest.java]: I refactored the methods a little bit

File: ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java (hunk 1 above)
Comment (MichaelsJP) [reply ors-api/src/test/java/org/heigit/ors/api/servlet/listeners/ORSInitContextListenerTest.java]: Added. But i left it as example.com

**Review by jhaeu — COMMENTED** [review]
Mthod SourceFileElements extractSourceFileElements(String sourceFilePropertyValue) works now as correct as before. But seems a little bit over complex to me. 
The task of this method is very simple, this is a little bit obfuscated now. To be honest, I found the old code much better readable.

**Review by jhaeu — APPROVED** [review]
LGTM

**Pull request merged**